crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -338,6 +338,7 @@ async fn cow_amm_jit(web3: Web3) {
 
     // Drive solution
     tracing::info!("Waiting for trade.");
+    onchain.mint_block().await;
     wait_for_condition(TIMEOUT, || async {
```

Comment:
I am not sure that `wait_for_condition` makes sense now without mining a new block on each iteration. Should we include the mining inside the loop? That would also avoid the requirement to mine a block manually each time before calling this function.